### PR TITLE
Update doc chapter "Adapter Methods"

### PR DESCRIPTION
**Pull request by TimoImhof**
This PR updates the Chapter "Adapter Methods" from the adapter docs.

Contributions:
1. Fix typos and inconsistencies
2. Update broken named anchors
3. Add more links to class and method documentation
4. Update broken example code

### CODE REVIEW

File: adapter_docs/overview.md [hunk 1]
```diff
@@ -30,32 +30,33 @@ While these adapters have laid the foundation of the adapter-transformers librar
...
 
 In the remaining sections, we will present how adapter methods can be configured in `adapter-transformers`.
-The next two pages will then present the methodological details of all currently supported adapter methods.
+The following two pages will offer the methodological details of all currently supported adapter methods.
 
 ## Table of Adapter Methods
 
 The following table gives an overview of all adapter methods supported by `adapter-transformers`.
 Identifiers and configuration classes are explained in more detail in the [next section](#configuration).
+TODO: update links
```
Comment (calpt) [inline adapter_docs/overview.md:44]: Still todo?

**Review by calpt — APPROVED** [review]
Thanks for fixing :)

File: adapter_docs/method_combinations.md [hunk 2]
```diff
@@ -77,7 +77,7 @@ $$
 In the configuration classes of `adapter-transformers`, these gating mechanisms can be activated via `use_gating=True`.
 The full UniPELT setup can be instantiated using `UniPELTConfig`[^unipelt]:
 
-[^unipelt]: Note that the implementation of UniPELT in `adapter-transformers` follows the implementation in the original code, which is slighlty different from the description in the paper. See [here](https://github.com/morningmoni/UniPELT/issues/1) for more.
+[^unipelt]: Note that the implementation of UniPELT in `adapter-transformers` follows the implementation in the original code, which is slightlty different from the description in the paper. See [here](https://github.com/morningmoni/UniPELT/issues/1) for more.
```
[the remark this answers is not in the document]
Comment (TimoImhof) [reply adapter_docs/method_combinations.md:80]: Thx for spotting

File: adapter_docs/overview.md (hunk 1 above)
Comment (TimoImhof) [reply adapter_docs/overview.md:44]: Not this is resolved, forgot to remove the todo, thanks for spotting

**Pull request merged**